### PR TITLE
GetIndexState only get the index state generated when CreateIndex is called

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #17460 
Signed-off-by: Cai.Zhang <cai.zhang@zilliz.com>

### CODE REVIEW

File: internal/rootcoord/meta_table.go [hunk 1]
```diff
@@ -936,6 +936,54 @@ func (mt *MetaTable) DropIndex(collName, fieldName, indexName string) (typeutil.
...
+	if !ok {
+		collID, ok = mt.collAlias2ID[collName]
+		if !ok {
+			return nil, fmt.Errorf("collection not exist with name = %s", collName)
+		}
+	}
+	collMeta, ok := mt.collID2Meta[collID]
+	if !ok {
+		return nil, fmt.Errorf("collection not exist with name = %s", collName)
+	}
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:953]: line `943~953` can be simplified by `mt.unlockGetCollectionInfo`.

File: internal/rootcoord/meta_table.go [hunk 2]
```diff
@@ -936,6 +936,54 @@ func (mt *MetaTable) DropIndex(collName, fieldName, indexName string) (typeutil.
...
+			continue
+		}
+		indexID = info.IndexID
+		break
+	}
+
+	if indexID == 0 {
+		log.Warn("get init buildIDs, index not found", zap.String("collection name", collName),
+			zap.String("index name", indexName))
+		return nil, fmt.Errorf("index not found with name = %s", indexName)
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:971]: Better to also add collection name in error.

File: internal/proto/etcd_meta.proto [hunk 3]
```diff
@@ -45,6 +45,7 @@ message SegmentIndexInfo {
   int64 indexID = 5;
   int64 buildID = 6;
   bool enable_index = 7;
+  bool by_flush = 8;
```
Comment (longjiquan) [inline internal/proto/etcd_meta.proto:48]: I wonder if `by_auto_flush` is better?

File: internal/rootcoord/meta_table.go [hunk 4]
```diff
@@ -936,6 +936,54 @@ func (mt *MetaTable) DropIndex(collName, fieldName, indexName string) (typeutil.
...
+	for _, indexID2Info := range mt.segID2IndexMeta {
+		segIndexInfo, ok := indexID2Info[indexID]
+		if !ok {
+			continue
+		}
+		if !segIndexInfo.ByFlush {
+			initBuildIDs = append(initBuildIDs, segIndexInfo.BuildID)
+		}
+	}
+	return initBuildIDs, nil
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:984]: Why we don't return all build ids when `create_index` was called? This is also asynchronouns.

File: internal/rootcoord/meta_table.go (hunk 1 above)
Comment (xiaocai2333) [reply internal/rootcoord/meta_table.go:946]: nice, I didn't this method.

File: internal/rootcoord/meta_table.go (hunk 2 above)
Comment (xiaocai2333) [reply internal/rootcoord/meta_table.go]: ok

File: internal/proto/etcd_meta.proto (hunk 3 above)
Comment (xiaocai2333) [reply internal/proto/etcd_meta.proto]: good idea.

File: internal/rootcoord/meta_table.go (hunk 4 above)
Comment (xiaocai2333) [reply internal/rootcoord/meta_table.go:977]: If sdk is down, user can't know what buildIDs there are.

**Pull request merged**